### PR TITLE
feat: add test, lint and image builder

### PR DESCRIPTION
**Pull request by YunxianHua**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -39,29 +38,29 @@ if(NOT CMAKE_BUILD_TYPE)
...
+else()
     message(STATUS "Using Boost ASIO")
     find_package(Boost REQUIRED)
-endif (USE_ASIO_STANDALONE)
+endif()
 
 include(TestBigEndian)
-TEST_BIG_ENDIAN(ENDIAN)
-if (ENDIAN)
+test_big_endian(ENDIAN)
```
Comment (zhexuany) [inline CMakeLists.txt:50]: Keep CAP case？

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+Any contribution that you make to this repository will
+be under the Apache 2 License, as dictated by that
+[license](http://www.apache.org/licenses/LICENSE-2.0.html):
+
+~~~
```
Comment (zhexuany) [inline CONTRIBUTING.md:5]: Maybe need delete this section?

File: cobridge_base/include/serialization.hpp [hunk 3]
```diff
@@ -12,69 +11,72 @@
...
 #ifdef ARCH_IS_BIG_ENDIAN
-        buf[0] = val & 0xff;
-        buf[1] = (val >> 8) & 0xff;
-        buf[2] = (val >> 16) & 0xff;
-        buf[3] = (val >> 24) & 0xff;
-        buf[4] = (val >> 32) & 0xff;
-        buf[5] = (val >> 40) & 0xff;
-        buf[6] = (val >> 48) & 0xff;
-        buf[7] = (val >> 56) & 0xff;
+  buf[0] = val & 0xff;
```
Comment (zhexuany) [inline cobridge_base/include/serialization.hpp:30]: align issue？

File: cobridge_base/include/serialization.hpp [hunk 4]
```diff
@@ -12,69 +11,72 @@
...
 
-    inline void write_uint32_LE(uint8_t *buf, uint32_t val) {
+inline void write_uint32_LE(uint8_t * buf, uint32_t val)
+{
 #ifdef ARCH_IS_BIG_ENDIAN
-        buf[0] = val & 0xff;
-        buf[1] = (val >> 8) & 0xff;
-        buf[2] = (val >> 16) & 0xff;
-        buf[3] = (val >> 24) & 0xff;
+  buf[0] = val & 0xff;
```
Comment (zhexuany) [inline cobridge_base/include/serialization.hpp:46]: align issue too?

File: CMakeLists.txt (hunk 1 above)
Comment (YunxianHua) [reply CMakeLists.txt:50]: ROS `ament_lint_cmake` did't allow mix upper and lower case commands, all other function use lower except this.

File: CONTRIBUTING.md (hunk 2 above)
Comment (YunxianHua) [reply CONTRIBUTING.md:5]: `CONTRIBUTING.md` is needed in ROS `ament_copyright`

File: cobridge_base/include/serialization.hpp (hunk 3 above)
Comment (YunxianHua) [reply cobridge_base/include/serialization.hpp:30]: yup, auto processed by ROS `ament_uncrustify`

File: cobridge_base/include/serialization.hpp (hunk 4 above)
Comment (YunxianHua) [reply cobridge_base/include/serialization.hpp:46]: just as above.

**Review by zhexuany — APPROVED** [review]
LGTM

**Pull request merged**